clippy_lints/src/casts/mod.rs

```diff
@@ -836,7 +836,7 @@ declare_clippy_lint! {
     /// ```
     #[clippy::version = "1.93.0"]
     pub NEEDLESS_TYPE_CAST,
-    nursery,
```

Comment:
I think we should keep the lint in nursery until more conclusive testing shows it's working as intended.